src/System.Runtime.Extensions/tests/System/IO/PathTests.cs

```diff
@@ -35,13 +35,28 @@ public static void ChangeExtension(string path, string newExtension, string expe
             Assert.Equal(expected, Path.ChangeExtension(path, newExtension));
         }
 
-        [Theory]
-        [InlineData("")]
-        [InlineData(" ")]
-        [InlineData("\r\n")]
-        public static void GetDirectoryName_EmptyOrWhitespace_Throws(string path)
+        [Fact]
+        public static void GetDirectoryName_EmptyThrows()
+        {
+            AssertExtensions.Throws<ArgumentException>("path", null, () => Path.GetDirectoryName(string.Empty));
+        }
+
+        [Theory,
+            InlineData(" "),
+            InlineData("\r\n")]
+        [ActiveIssue(21358)] // Pending CoreCLR behavior change
+        public static void GetDirectoryName_SpaceOrControlCharsThrowOnWindows(string path)
         {
-            AssertExtensions.Throws<ArgumentException>("path", null, () => Path.GetDirectoryName(path));
+            Action action = () => Path.GetDirectoryName(path);
+            if (PlatformDetection.IsWindows)
```

Comment:
I don't agree in this case. This particular case is saying "throws on Windows but not Unix". Same input data. I could make the test name longer, but I think the fact that it isn't Windows only makes it implicit that it is "WindowsAndNotUnix".

I could split this into two entry points and shared test data property, but that seems like over-kill to me in this case. Same goes for duplicating the method.  I think it is more coherent to have both sides of the coin together here.